File: src/main/java/src/Person.java
public class Person {
    Integer id;
    String name;
    String phone;
    Integer age;

    public Person(Integer id, Integer age, String name, String phone) {
        this.id = id;
        this.name = name;
        this.phone = phone;
        this.age = age;
    }

    @Override
    public String toString() {
        return "Person{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", phone='" + phone + '\'' +
                ", age=" + age +
                '}';
    }
}

File: src/main/kotlin/person.kt

data class Person(
    val id: Int? = null,
    val name :String? = null,
    val age: Int? = null,
    val phone: String? = null,
){
   val truePhone: String?
        get() = this.phone.toPhone()

}

fun String?.toPhone():String{
    return if (this == null) "не задан телефон"
    else if (this.length != 11) this
    else {

        val el2 = this.substring(startIndex = 1, endIndex = 4)
        val el3 = this.substring(startIndex = 4, endIndex = 7)
        val el4 = this.substring(startIndex = 7, endIndex = 9)
        val el5 = this.substring(startIndex = 9,)
        "+7($el2)-$el3-$el4-$el5"
    }
}